addon/components/search-field.js

```diff
@@ -12,17 +12,24 @@ import { action } from '@ember/object';
   @public
 */
 export default class SearchFieldComponent extends Component {
-  @action handleInputEvent(e) {
-    this.setValue(e.target.value);
+  @action
+  handleInputEvent(e) {
+    this.safeTrigger('onChange', e.target.value);
   }
 
-  @action clear() {
-    this.setValue('');
+  @action
+  handleSubmit(e) {
+    e.preventDefault();
+
+    this.safeTrigger('onSubmit', e.target.querySelector('input').value);
+  }
+
+  @action
+  clear() {
+    this.safeTrigger('onChange', '');
   }
 
-  setValue(value) {
-    if (this.args.onChange) {
-      this.args.onChange(value);
-    }
+  safeTrigger(callback, value) {
+    if (this.args[callback]) this.args[callback](value);
```

Comment:
Is this like the only way in Ember to achieve this? I'm asking out of curiosity, because in Vue you can use the events system and do:

```js
safeTrigger(value) {
  this.$emit('submit', value);
}
```

and

```html
<Search @submit="doMagic" />
```